File: packages/emitter/testcases/common/types/function-collections/FunctionArrays.ts
import { int } from "@tsonic/core/types.js";

export type Operation = (a: int, b: int) => int;

// Array of functions
export const operations: Operation[] = [
  (a, b) => a + b,
  (a, b) => a - b,
  (a, b) => a * b,
];

// Interface with function properties
export interface OperationMap {
  add: Operation;
  subtract: Operation;
  multiply: Operation;
}

File: packages/emitter/testcases/common/expected/types/function-collections/FunctionArrays.cs
namespace TestCases.common.types.functioncollections
{
    public class OperationMap
    {
        public required global::System.Func<int, int, int> add { get; set; }

        public required global::System.Func<int, int, int> subtract { get; set; }

        public required global::System.Func<int, int, int> multiply { get; set; }
    }

    [global::Tsonic.Internal.ModuleContainerAttribute]
    public static class FunctionArrays
    {
        public static readonly global::System.Func<int, int, int>[] operations = new global::System.Func<int, int, int>[] { (int a, int b) => a + b, (int a, int b) => a - b, (int a, int b) => a * b };
    }
}